### ISSUE TITLE
Dark Forces can't find Sound Card & no Sound

### ISSUE THREAD

**Issue by adamcirillo** [luciusDXL/TheForceEngine#183]
Version 1.01
[the_force_engine_log.txt](https://github.com/luciusDXL/TheForceEngine/files/10367639/the_force_engine_log.txt)

game is playing midi music but getting no sound.
sound card is sound blaster ZXR.
running Windows 11 22h2 with latest updates & drivers

**Comment by luciusDXL** [comment]
For people having audio problems using TFE - the issue is that there are multiple possible APIs that can be used, and dedicated sound cards don't support them all well. TFE uses the Windows suggested default (WASAPI), but that doesn't always work. So the next release will not only do a much better job reporting errors, to help debugging audio issues, but will also attempt to use different audio APIs when failures happen (including ASIO and Direct Sound) - which should greatly increase support. 
In addition, once the system is properly initialized, you will be able to select your output device - in case the default isn't desired for some reason.
So hopefully this will resolve the audio issues that people are having. 🙂

**Comment by luciusDXL** [comment]
Please test with the version 1.02 Release.

**Comment by adamcirillo** [comment]
I just tried 1.02 still got no devices picked up at all here is the log

[Main] The Force Engine 
v1.02.000-1-g7ec69bb1

[Paths] Program Path: "G:\Steam\steamapps\common\Dark Forces\"
[Paths] Program Data: "C:\ProgramData\TheForceEngine\"
[Paths] User Documents: "C:\Users\adam.ADAMCIRILLO\Documents\TheForceEngine\"
[Paths] Source Data: "G:/Steam/steamapps/common/Dark Forces/Game/"
[Startup] TFE_System::init
[Display] Fullscreen enabled.
[RenderBackend] OpenGL Device Tier: 3
[Startup] TFE_AudioSystem::init
[Audio] Attempting to initialize Audio System using API 'Windows WASAPI'.
[Warning : Audio] Cannot query audio output devices. Error: 'RtApiWasapi::getDeviceInfo: Unable to retrieve device mix format.', type: 'DRIVER_ERROR'
[Audio] Attempting to initialize Audio System using API 'Windows ASIO'.
[Warning : Audio] Cannot query audio output devices. Error: 'RtApiWasapi::getDeviceInfo: Unable to retrieve device mix format.', type: 'DRIVER_ERROR'
[Audio] Attempting to initialize Audio System using API 'Windows DirectSound'.
[Warning : Audio] Cannot query audio output devices. Error: 'RtApiWasapi::getDeviceInfo: Unable to retrieve device mix format.', type: 'DRIVER_ERROR'
[Error : Audio] Cannot initialize audio device.
[Error : Audio] Cannot start audio device.
[Startup] TFE_MidiPlayer::init
[Startup] TFE_Polygon::init
[Startup] TFE_Image::init
[Startup] TFE_FrontEndUI::init
[MemoryRegion] Allocated new memory block in region 'game' - new size is 1 blocks, total size is '8388608'
[MemoryRegion] Allocated new memory block in region 'level' - new size is 1 blocks, total size is '8388608'
[Progam Flow] The Force Engine Game Loop Started
[Audio] Shutdown
[MidiPlayer] Shutdown
[Progam Flow] The Force Engine Game Loop Ended.

**Comment by mlauss2** [comment]
TFE probably needs an update of RtAudio itself, the shipped version is old by now, and there were a few updates to the windows internals by now.

**Comment by adamcirillo** [comment]
I finally figured out the problem it was my wireless headphones causing the problem when I removed the receiver the sound card showed up. after doing some googling EPOS recommend removing the driver and install the Microsoft one and not theirs and then it worked with the headphones plugged in. I was having audio issues with other games and that lead me to believe the problem was with my system and not the The Force Engine.
